src/mysql-util/src/schemas.rs

```diff
@@ -309,5 +254,185 @@ where
+                    qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                    column_name: info.column_name.clone(),
+                    intended_type: None,
+                })?,
+        }),
+        "text" | "mediumtext" | "longtext" => Ok(ScalarType::String),
+        "binary" | "varbinary" | "tinyblob" | "blob" | "mediumblob" | "longblob" => {
+            Ok(ScalarType::Bytes)
+        }
+        _ => Err(UnsupportedDataType {
+            column_type: info.column_type.clone(),
+            qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+            column_name: info.column_name.clone(),
+            intended_type: None,
+        }),
+    }
+}
+
+/// Parse the specified column as a TEXT COLUMN. We only support the set of types that are
+/// represented as an encoded-string in both the mysql-common binary query response and binlog
+/// event representation, OR types that we've added explicit casting support for.
+fn parse_as_text_column(
+    info: &InfoSchema,
+    schema_name: &str,
+    table_name: &str,
+) -> Result<(ScalarType, Option<MySqlColumnMeta>), UnsupportedDataType> {
+    match info.data_type.as_str() {
+        "json" => Ok((ScalarType::String, Some(MySqlColumnMeta::Json))),
+        "enum" => Ok((
+            ScalarType::String,
+            Some(MySqlColumnMeta::Enum(MySqlColumnMetaEnum {
+                values: enum_vals_from_column_type(info.column_type.as_str()).map_err(|_| {
+                    UnsupportedDataType {
+                        column_type: info.column_type.clone(),
+                        qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+                        column_name: info.column_name.clone(),
+                        intended_type: Some("text".to_string()),
+                    }
+                })?,
+            })),
+        )),
+        _ => Err(UnsupportedDataType {
+            column_type: info.column_type.clone(),
+            qualified_table_name: format!("{:?}.{:?}", schema_name, table_name),
+            column_name: info.column_name.clone(),
+            intended_type: Some("text".to_string()),
+        }),
+    }
+}
+
+static ENUM_VAL_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"'((?:[^']|'')*)'").expect("valid regex"));
+
+/// Parse the enum values from a column_type value on an enum column, which is a string like
+/// "enum('apple','banana','cher,ry','ora''nge')"
+/// We need to handle the case where the enum value itself contains a comma or a
+/// single quote (escaped with another quote), so we use a regex to do so
+fn enum_vals_from_column_type(s: &str) -> Result<Vec<String>, anyhow::Error> {
+    let vals_str = s
+        .strip_prefix("enum(")
+        .and_then(|s| s.strip_suffix(')'))
+        .ok_or(anyhow::format_err!(
+            "Unable to parse enum column type string"
+        ))?;
+
+    Ok(ENUM_VAL_REGEX
+        .captures_iter(vals_str)
+        .map(|s| s[1].to_string())
```

Comment:
There is a missing step here to undo the single quote escaping here. In the test below the enum value `"ora'nge"` is incorrectly parsed as `"ora''nge"`. Are there other escape sequences that we need to worry about? e.g tabs or other weird things? 

```suggestion
        .map(|s| s[1].replace("''", "'"))
```

Reply:
good catch! As far as I can tell the only tricky characters to worry about are commas and the escaped single-quotes. The docs are not very helpful in describing this unfortunately